pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -126,6 +134,10 @@ def advance(self, *args: Any, **kwargs: Any) -> None:
         Raises:
             StopIteration: When the epoch is canceled by the user returning -1
         """
+        # used to jump to the validation_loop on `advance_end`.
+        if self.restarting and self._should_check_val_fx(self.batch_idx, self.is_last_batch):
+            return
```

Comment:
Why do we want to make this jump? Can it be avoided?

Reply:
I don't think it can be avoided. 
We want to make this jump to prevent fetching a batch when val_check_interval is set and the loop failed during the validation loop. We should jump directly the on_advanced_end.